### PR TITLE
Save post number in ImageLink, for ease of use in GenerateNewFilename

### LINKED ISSUE #28: Save post number in ImageLink, for ease of use in GenerateNewFilename
Couple of things:
- Give `Tim` property some documentation, explain what it is by reading the documentation here: https://github.com/4chan/4chan-API/blob/master/pages/Threads.md
- 1 location still uses the old constructor, remove the old constructor and update the calling location to use the new one.
- Update the now only constructor to take `no` as a parameter, save it in a new public property (give the new property some good documentation too).
- Rename `GenerateNewFilename` method to `GenerateFilename`.
- Rename `URL` property to `Url` (don't forget to update the `ToString` too).

### PR DESCRIPTION
**Pull request by royaljackal**
For the 8kun currently post id is being saved into the Tim variable as the actual tim is a string of some sort. Maybe Tim variable should be made into string so both 8kun and 4chan tims would be stored as strings?

### CODE REVIEW

File: GChan/Trackers/Sites/Thread_4Chan.cs [hunk 1]
```diff
@@ -64,7 +64,8 @@ protected override ImageLink[] GetImageLinks()
                     .Select(x =>
                         new ImageLink(long.Parse(x[timPath].ToString()),
                             baseURL + x[timPath] + x["ext"],
-                            x["filename"].ToString()))
+                            x["filename"].ToString(),
+                            long.Parse(x["no"].ToString())))
```
Comment (Issung) [inline GChan/Trackers/Sites/Thread_4Chan.cs]: Try doing `x["no"].Value<long>()` here and in other locations, test and see if it works as expected.

**Review by Issung — CHANGES_REQUESTED** [review]
1 small change try to get the value in the intended type from the JTokens to avoid unecessary string allocations and more descriptive error messages.
Few things to think about for this class, as its behaviour is a bit different for each website.. But this was intended to be a small issue/PR so let's keep it that way.

Also please add your name to `AboutBox.cs` in the list of contributors :)

**Review by Issung — APPROVED** [review]
Looking great :)

**Comment by Issung** [comment]
Merge when you're ready :)

**Comment by Issung** [comment]
Are you ready to merge this PR?

**Comment by royaljackal** [comment]
Yeah, merge it whenever you like.

**Review by Issung — APPROVED** [review]

**Pull request merged**